spec/integ/crypto/megolm-backup.spec.ts

```diff
@@ -70,101 +78,133 @@ const CURVE25519_BACKUP_INFO: IKeyBackupInfo = {
-                        timeline: {
-                            events: [ENCRYPTED_EVENT],
-                        },
-                    },
-                },
+    async function initTestClient(opts: Partial<ICreateClientOpts> = {}): Promise<MatrixClient> {
+        const client = createClient({
+            baseUrl: TEST_HOMESERVER_URL,
+            userId: TEST_USER_ID,
+            accessToken: "akjgkrgjs",
+            deviceId: TEST_DEVICE_ID,
+            ...opts,
+        });
+        await client.initCrypto();
+
+        return client;
+    }
+
+    it("Alice checks key backups when receiving a message she can't decrypt", async function () {
+        fetchMock.get("express:/_matrix/client/v3/room_keys/keys/:room_id/:session_id", CURVE25519_KEY_BACKUP_DATA);
+        fetchMock.post("express:/_matrix/client/r0/keys/upload", {
+            one_time_key_counts: {
+                curve25519: 100,
+                signed_curve25519: 100,
             },
-        };
-
-        return aliceTestClient!
-            .start()
-            .then(() => {
-                return createOlmSession(testOlmAccount, aliceTestClient);
-            })
-            .then(() => {
-                const privkey = decodeRecoveryKey(RECOVERY_KEY);
-                return aliceTestClient!.client!.crypto!.storeSessionBackupPrivateKey(privkey);
-            })
-            .then(() => {
-                aliceTestClient!.httpBackend.when("GET", "/sync").respond(200, syncResponse);
-                aliceTestClient!.expectKeyBackupQuery(ROOM_ID, SESSION_ID, 200, CURVE25519_KEY_BACKUP_DATA);
-                return aliceTestClient!.httpBackend.flushAllExpected();
-            })
-            .then(function (): Promise<MatrixEvent> {
-                const room = aliceTestClient!.client.getRoom(ROOM_ID)!;
-                const event = room.getLiveTimeline().getEvents()[0];
-
-                if (event.getContent()) {
-                    return Promise.resolve(event);
-                }
-
-                return new Promise((resolve, reject) => {
-                    event.once(MatrixEventEvent.Decrypted, (ev) => {
-                        logger.log(`${Date.now()} event ${event.getId()} now decrypted`);
-                        resolve(ev);
-                    });
-                });
-            })
-            .then((event) => {
-                expect(event.getContent()).toEqual("testytest");
+        });
+        // mock for the outgoing key requests that will be sent
+        fetchMock.put("express:/_matrix/client/r0/sendToDevice/m.room_key_request/:txid", {});
+
+        fetchMock.get("express:/_matrix/client/v3/room_keys/version", CURVE25519_BACKUP_INFO);
+
+        aliceClient = await initTestClient();
+
+        // we need the backup to be trusted for the test to work
+        const backupDataToSign = JSON.parse(JSON.stringify(CURVE25519_BACKUP_INFO));
+
+        await aliceClient.crypto!.signObject(backupDataToSign.auth_data);
```

Comment:
I feel like we should find out why this is needed when it wasn't before.

Reply:
The legacy test was forcing:
> aliceTestClient!.client.crypto!.backupManager.backupInfo = CURVE25519_BACKUP_INFO;

This was by-passing the checkAndStart() that is supposed to download and check (and would fail with the incomplete test data). `backupInfo` is only set if the backup is trusted